### PR TITLE
chore(kuma-cp) change exec probes to http

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
# Summary

This PR changes the way we probe Kuma DP sidecar.
So far we were using exec healthchecks and `/ready` endpoint on Admin API. Admin API is exposed only on loopback interface for security reasons.

This PR exposes `/ready` to the outside of the container, but only the `/ready` endpoint. This way we can change probe from exec to http

# Reason

Containerd 1.4.0 had a bug of Pods being stuck at Terminating status when there was a container with exec probe.
https://github.com/awslabs/amazon-eks-ami/issues/563

### CODE REVIEW

File: pkg/xds/generator/admin_proxy_generator.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+import (
+	core_xds "github.com/kumahq/kuma/pkg/core/xds"
+	xds_context "github.com/kumahq/kuma/pkg/xds/context"
+	envoy_clusters "github.com/kumahq/kuma/pkg/xds/envoy/clusters"
+	envoy_listeners "github.com/kumahq/kuma/pkg/xds/envoy/listeners"
+	envoy_names "github.com/kumahq/kuma/pkg/xds/envoy/names"
+)
+
+// OriginPrometheus is a marker to indicate by which ProxyGenerator resources were generated.
```
Comment (nickolaev) [inline pkg/xds/generator/admin_proxy_generator.go:11]: nit: the comment is about OriginPrometheus -> OriginAdmin

**Review by nickolaev — APPROVED** [review]
Looks great to me!

**Comment by nickolaev** [comment]
Shall we backport?

File: pkg/xds/generator/admin_proxy_generator.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+			return nil, err
+		}
+		resources.Add(&core_xds.Resource{
+			Name:     listener.GetName(),
+			Origin:   OriginAdmin,
+			Resource: listener,
+		})
+	}
+
+	resources.Add(&core_xds.Resource{
```
Comment (lobkovilya) [inline pkg/xds/generator/admin_proxy_generator.go:62]: Do we need this cluster if `proxy.Dataplane.Spec.GetNetworking().Address == "127.0.0.1"`?

Comment (jakubdyszkiewicz) [reply pkg/xds/generator/admin_proxy_generator.go:62]: Yes, because we may still use it for Prometheus

**Review by lobkovilya — APPROVED** [review]

**Pull request merged**